sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -651,7 +755,34 @@ def get_media_settings_value(physical_port, key):
     return {}
 
 
-def get_media_settings_key(physical_port, transceiver_dict):
+def get_speed_and_lane_count(port, cfg_port_tbl):
+    port_speed, lane_count = '0', '0'
+    found, port_info = cfg_port_tbl.get(port)
+    port_info_dict = dict(port_info)
+    if found and 'speed' in port_info_dict and 'lanes' in port_info_dict:
+        port_speed = port_info_dict['speed']
+        lanes = port_info_dict['lanes']
+        lane_count = len(lanes.split(','))
+    return port_speed, lane_count
+
+
+def get_lane_speed_key(physical_port, port_speed, lane_count):
+    sfp = platform_chassis.get_sfp(physical_port)
+    api = sfp.get_xcvr_api()
+    
+    lane_speed_key = None
+    if is_cmis_api(api):
+        appl_adv_dict = api.get_application_advertisement()
+        app_id = get_cmis_application_desired(api, int(lane_count), int(port_speed))
+        if app_id:
+            host_electrical_interface_id = appl_adv_dict[app_id].get('host_electrical_interface_id')
```

Comment:
please check that app_id is in appl_adv_dict

Reply:
I don't think it is possible to have an app_id that is both not None and not have it in appl_adv_dict, because it is taken from that dictionary from the first place, but I'll add that just to be on the safe side.